src/Core/src/Platform/iOS/KeyboardAutoManagerScroll.cs

```diff
@@ -302,10 +292,23 @@ internal static void AdjustPosition()
 
 		var topLayoutGuide = Math.Max(navigationBarAreaHeight, ContainerView.LayoutMargins.Top) + 5;
 
-		var keyboardYPosition = window.Frame.Height - kbSize.Height - TextViewTopDistance;
+		// calculate the cursor rect
+		var localCursor = FindLocalCursorPosition();
+		if (localCursor is CGRect local)
+			CursorRect = View.ConvertRectToView(local, null);
+
+		if (CursorRect is null)
+		{
+			IsKeyboardAutoScrollHandling = false;
+			return;
+		}
 
 		var viewRectInWindow = View.ConvertRectToView(View.Bounds, window);
 
+		TextViewTopDistance = ((int?)localCursor?.Height ?? 0) + 20;
```

Comment:
Could you help me understand what the `+20` offset is? I assume this was viewed in the previous PR, but I'm curious